addons/survey/static/src/js/survey_form.js

```diff
@@ -26,7 +26,12 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
         this.options = this.$target.find('form').data()
         var $timer = $('.o_survey_timer');
         if ($timer.length) {
-            this.surveyTimerWidget = new publicWidget.registry.SurveyTimerWidget(this);
+            var timeLimitMinutes = this.options['timeLimitMinutes'];
+            var timer = this.options['timer'];
+            this.surveyTimerWidget = new publicWidget.registry.SurveyTimerWidget(this, {
+                'timer': timer,
+                'timeLimitMinutes': timeLimitMinutes
+            });
             this.surveyTimerWidget.attachTo($timer);
             this.surveyTimerWidget.on('time_up', this, function (ev) {
                 self.$el.find('button[name="button_submit"]').click();
```

Comment:
I think you can do the binding in a more standard way now:

- Add a custom event 'time_up' on the survey widget which would be bound to a method that would do the click submit
- In the timerWidget, bind its own 'time_up' event that would do something like this.trigger_up('time_up').